### PR TITLE
feat(frontend): authorization frontend

### LINKED ISSUE #370: 🚀 [Feature] Protect Frontend Routes
## 🚀 Feature

Protect all frontend routes so that the frontend is just available when logged in.

The idea is to allow only authorized users to access the frontend.

- [ ] Allow Access when auth cookie is available
- [ ] Redirect to presenter when auth cookie is not present
- [ ] Investigate: Define Issue what software parts should be moved from Presenter to Backend and how routing should take place. Furthermore define a goal for authentik to support this new architecture.

### PR DESCRIPTION
**Pull request by Mogge**
## 🍰 Pullrequest

Based on #403 

### Issues
- fixes #370

### Todo

- [X] None

### CODE REVIEW

**Review by Elweyn — COMMENTED** [review]
Seems good but could not test it...

**Review by Elweyn — DISMISSED** [review]
I can't let it run with docker but if I change ``presenter/server/index.ts`` to default port 3001 then I test it with:
- cd authentik && docker compose up
- cd frontend && npm run dev || prod
- cd presenter && npm run dev || prod

The logic works fine

**Review by ulfgebhardt — CHANGES_REQUESTED** [review]
image

The guard throws serious errors - I am fine with merging this and resolve it after - but there is a problem.

**Review by ulfgebhardt — COMMENTED** [review]
The page /signin contains the signout button. This button must be reactive to the logged in state and not visible when not logged in (can be postponed)

image

**Review by ulfgebhardt — COMMENTED** [review]
image

I think we should require correct values for this - maybe we want so set a default here, since the frontend only makes sense to operate when authentication is possible.
Furthermore the default values for

```
 AUTHORITY: (import.meta.env.PUBLIC_ENV__AUTH__AUTHORITY ?? '') as string,
  AUTHORITY_SIGNUP_URI: (import.meta.env.PUBLIC_ENV__AUTH__AUTHORITY_SIGNUP_URI ?? '') as string,
```
make no sense anymore, since the presenter hides its buttons, not the frontend

Furthermore `AUTHORITY_SIGNUP_URI` is not needed in the frontend.

File: presenter/src/components/menu/HeaderMenu.vue [hunk 1]
```diff
@@ -147,15 +147,17 @@ const authService = inject<AuthService>('authService')
...
-// TODO what about the store?
 async function signIn() {
-  try {
-    await authService?.signIn()
-    navigate('/')
-  } catch (error) {
-    // eslint-disable-next-line no-console
-    console.log('auth error', error)
-  }
+  window.location.href = AUTH.SIGNIN_REDIRECT_URI
```
[the remark this answers is not in the document]
Comment (Mogge) [reply presenter/src/components/menu/HeaderMenu.vue:151]: Yes

File: presenter/src/components/menu/HeaderMenu.vue [hunk 2]
```diff
@@ -147,15 +147,17 @@ const authService = inject<AuthService>('authService')
...
+  /*
+   try {
+     await authService?.signIn()
+     navigate('/')
+   } catch (error) {
+     // eslint-disable-next-line no-console
+     console.log('auth error', error)
+   }
+   */
```
[the remark this answers is not in the document]
Comment (Mogge) [reply presenter/src/components/menu/HeaderMenu.vue:160]: Is removed in the next PR

**Comment by ulfgebhardt** [comment]
> image
> 
> I think we should require correct values for this - maybe we want so set a default here, since the frontend only makes sense to operate when authentication is possible. Furthermore the default values for
> 
> ```
>  AUTHORITY: (import.meta.env.PUBLIC_ENV__AUTH__AUTHORITY ?? '') as string,
>   AUTHORITY_SIGNUP_URI: (import.meta.env.PUBLIC_ENV__AUTH__AUTHORITY_SIGNUP_URI ?? '') as string,
> ```
> 
> make no sense anymore, since the presenter hides its buttons, not the frontend
> 
> Furthermore `AUTHORITY_SIGNUP_URI` is not needed in the frontend.

Fixed - I kept the `AUTHORITY_SIGNUP_URI` since that would cripple the authService if not present.

**Review by ulfgebhardt — CHANGES_REQUESTED** [review]
Still doesn't work in production.

While I can see it working in dev, I don't get it to work on production.

Running on dev I can see the redirect happening on console and browser.

```
npm install
npm run dev
```

image
image
image

In production nothing happens neither on console nor in the the browser

```
npm install
npm build
npm run server:prod
```

image

**Comment by Mogge** [comment]
Looks like a build problem to me. The guard does not trigger at all all when building for production. If I remove all the code and just leave a console, no output is seen anywhere

**Review by ulfgebhardt — APPROVED** [review]

**Pull request merged**